cluster/src/dunit/scala/io/snappydata/cluster/CassandraSnappyDUnitTest.scala

```diff
@@ -0,0 +1,223 @@
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.cluster
+
+import java.io.{BufferedWriter, ByteArrayInputStream, File, FileFilter, FileOutputStream, OutputStreamWriter, PrintWriter}
+import java.util
+
+import scala.sys.process._
+import scala.language.postfixOps
+
+import io.snappydata.test.dunit.DistributedTestBase
+import io.snappydata.test.dunit.DistributedTestBase.WaitCriterion
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.filefilter.{IOFileFilter, TrueFileFilter, WildcardFileFilter}
+
+import org.apache.spark.{Logging, TestPackageUtils}
+
+class CassandraSnappyDUnitTest(val s: String)
+    extends ClusterManagerTestBase(s)
+        with Logging {
+  // scalastyle:off println
+
+  private val snappyProductDir = System.getenv("SNAPPY_HOME")
+
+  val scriptPath = s"$snappyProductDir/../../../cluster/src/test/resources/scripts"
+
+  val currDir = System.getProperty("user.dir")
+
+  val userHome = System.getProperty("user.home")
+
+  var cassandraClusterLoc = ""
+
+  private val commandOutput = "command-output.txt"
+
+  def snappyShell: String = s"$snappyProductDir/bin/snappy-sql"
+
+  override def beforeClass(): Unit = {
+
+    logInfo(s"Starting snappy cluster in $snappyProductDir/work")
+
+    logInfo((snappyProductDir + "/sbin/snappy-start-all.sh").!!)
+
+    val start = System.currentTimeMillis
+    "curl -OL http://www-us.apache.org/dist/cassandra/2.1.21/apache-cassandra-2.1.21-bin.tar.gz".!!
+    val end1 = System.currentTimeMillis
+    logInfo("Time to download cassandra " + (end1 - start))
+    logInfo("Cassandra jar downloaded")
+    val jarLoc = getUserAppJarLocation("apache-cassandra-2.1.21-bin*", currDir)
+    ("tar xvf " + jarLoc).!!
+    logInfo("Cassandra jar Unpackaed")
+    val cmd = Seq("find", currDir, "-name", "apache-cassandra-2.1.21", "-type", "d")
+    val res = cmd.lineStream_!.toList
+    logInfo("Cassandra folder location : " + res)
+    cassandraClusterLoc = res.head
+    (cassandraClusterLoc + "/bin/cassandra").!!
+    logInfo("Starting cassandra cluster " + cassandraClusterLoc)
+    val end = System.currentTimeMillis
+    logInfo("Time to install and start cassandra cluster " + (end - start))
+
+  }
+
+  override def afterClass(): Unit = {
+
+    logInfo(s"Stopping snappy cluster in $snappyProductDir/work")
+    logInfo((snappyProductDir + "/sbin/snappy-stop-all.sh").!!)
+
+    /* FileUtils.moveDirectory(new File(s"$snappyProductDir/work"), new File
+    (s"$snappyProductDir/workTestCassandraSnappy")) */
+    s"rm -rf $snappyProductDir/work".!!
+
+    logInfo("Stopping cassandra cluster")
+    val p = Runtime.getRuntime.exec("pkill -f cassandra")
+    p.waitFor()
+    p.exitValue() == 0
+    logInfo("Cassandra cluster stopped successfully")
+  }
+
+  def getJobJar(className: String, packageStr: String = ""): String = {
```

Comment:
These 3/4 methods are similar in to those in SplitClusterDUnitSecurityTest. May be you can refactor a bit and reuse.